Patch by Kurt Lust for the LUMI consortium
Adapt the MesonNinja EasyBlock to work with our buildtools module.
diff -Nau easybuild_easyblocks-4.8.2/easybuild/easyblocks/generic/mesonninja.py.orig easybuild_easyblocks-4.8.2/easybuild/easyblocks/generic/mesonninja.py
--- easybuild-easyblocks-4.8.2/easybuild/easyblocks/generic/mesonninja.py.orig	2025-11-24 18:53:46.000000000 +0200
+++ easybuild-easyblocks-4.8.2/easybuild/easyblocks/generic/mesonninja.py	2025-11-24 18:54:49.000000000 +0200
@@ -66,7 +66,8 @@
         # make sure both Meson and Ninja are included as build dependencies
         build_dep_names = [d['name'] for d in self.cfg.builddependencies()]
         for tool in ['Ninja', 'Meson']:
-            if tool not in build_dep_names:
+            #if tool not in build_dep_names:
+            if get_software_version(tool) is None:
                 raise EasyBuildError("%s not included as build dependency", tool)
             cmd = tool.lower()
             if not which(cmd):
